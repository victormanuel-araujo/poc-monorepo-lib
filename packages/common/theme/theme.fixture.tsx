import React from 'react'

// import { render } from '@testing-library/react-native'
// import { ThemeProvider } from 'styled-components/native'

// import { createTheme } from './'

// export function ThemeFixture({ children }) {
//   return <ThemeProvider theme={createTheme()}>{children}</ThemeProvider>
// }

// export function renderWithTheme(element) {
//   return render(<ThemeFixture>{element}</ThemeFixture>)
// }
