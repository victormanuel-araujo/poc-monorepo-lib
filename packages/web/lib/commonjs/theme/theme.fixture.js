import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createTheme } from "./index";
export function ThemeFixture({
  children
}) {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: createTheme()
  }, children);
}
export function renderWithTheme(element) {
  return render( /*#__PURE__*/React.createElement(ThemeFixture, null, element));
}
//# sourceMappingURL=theme.fixture.js.map