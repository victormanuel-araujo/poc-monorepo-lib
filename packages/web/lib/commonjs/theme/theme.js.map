{"version":3,"sources":["theme.ts"],"names":["colors","fonts","fontSizes","fontWeights","lineHeights","paddings","units","theme"],"mappings":"AACA,SAASA,MAAT,EAAiBC,KAAjB,EAAwBC,SAAxB,EAAmCC,WAAnC,EAAgDC,WAAhD,EAA6DC,QAA7D,EAAuEC,KAAvE;AACA;AACA;AACA;AACA;;AAEA;AACA;AACA;;AACA,OAAO,MAAMC,KAAY,GAAG;AAC1BP,EAAAA,MAD0B;AAE1BC,EAAAA,KAF0B;AAG1BC,EAAAA,SAH0B;AAI1BC,EAAAA,WAJ0B;AAK1BC,EAAAA,WAL0B;AAM1BE,EAAAA,KAN0B;AAO1BD,EAAAA;AAP0B,CAArB","sourcesContent":["import { Theme } from './theme.types'\nimport { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'\n/*\n * in the future, if exists a second theme, just have to create a second\n * composition as same is `theme`.\n */\n\n/**\n * Default gama theme - SMASH\n */\nexport const theme: Theme = {\n  colors,\n  fonts,\n  fontSizes,\n  fontWeights,\n  lineHeights,\n  units,\n  paddings,\n}\n"]}