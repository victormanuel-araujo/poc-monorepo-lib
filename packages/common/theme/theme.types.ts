import { ThemedStyledProps } from 'styled-components'

import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
export { ThemeProps } from 'styled-components/native'

// definitions of theme types

/**
 * the name of the colors of theme
 */
export type ThemeColorsName = 'primary' | 'secondary' | 'black' | 'white' | 'gray' | 'red' | 'transparent' | 'yellow'

/**
 * Existing tones for each color (ThemeColorsName). Each one color must have the
 * tone "base" and optionally the "light", "lighter", "dark" and "darker" tones
 */
export interface ThemeColorsShade {
  base: string
  light?: string
  lighter?: string
  dark?: string
  darker?: string
}

/**
 * The group of the colors (ThemeColorsName) and shades (ThemeColorsShade)
 *
 * @example
 * Each one color in ThemeColorsName becomes a key of an object, this object must have
 * the shades of ThemeColorsShade
 * ```json
 * {
 *   primary: {
 *     base: '#fff',
 *     light: '#fff',
 *     lighter: '#fff',
 *   }
 * }
 * ```
 */
export type ThemeColorsGroup = { [color in ThemeColorsName]: ThemeColorsShade }

/**
 * Final type of theme colors, bases on colors defined before.
 * that is: Union of ThemeColorsName and ThemeColorsShades with their values defined
 */
export type ThemeColors = typeof colors

/**
 * Type created based on values of font families defined on theme creation.
 * That is: If a family needs to be adde or removed a family, the type are
 * updated automatically
 */
export type ThemeFonts = { [fontName in keyof typeof fonts]: string }
/**
 * Type created based on font sizes also defined when creating the theme
 * @example
 * Sizes defined as follows:
 * ```json
 * {
 *   sm: '0px' as '0px',
 *   md: '0px' as '0px',
 *   ...
 * }
 * ```
 */
export type ThemeFontSizes = { [fontSize in keyof typeof fontSizes]: number }

/**
 * Type of font weights, based on values defined also in the theme creation
 * @example
 * Weights defined as follows:
 * ```json
 * {
 *   regular: 500 as 500,,
 *   bold: 900 as 900,
 *   ...
 * }
 * ```
 */
export type ThemeFontWeights = { [fontSize in keyof typeof fontWeights]: number }

/**
 * Type of the line heights, based on their values defined
 */
export type ThemeLineHeights = { [fontSize in keyof typeof lineHeights]: number }

/**
 * Type created based on measure units values defineds on theme creation
 * @example
 * Any units as example:
 * - base unit of spacing: 8
 * - header height (mobile): 60
 */
export type ThemeUnits = { [fontSize in keyof typeof units]: number }

/**
 * Type of pre-definitions of paddings (using base unit). Pre-definition also created
 * in initial theme values.
 * @example
 * The padding values are defined using the base unit, that is:
 * ```json
 * {
 *   sm: units.base as 8,
 *   md: (units.base * 2) as 16,
 *   lg: (units.base * 3) as 24,
 *   ...
 * }
 * ```
 */
export type ThemePaddings = { [fontSize in keyof typeof paddings]: number }

/**
 * The final interface of the theme.
 * The union of all types/interfaces created before.
 */
export interface Theme {
  colors: ThemeColors

  fonts: ThemeFonts
  fontSizes: ThemeFontSizes
  fontWeights: ThemeFontWeights
  lineHeights: ThemeLineHeights

  units: ThemeUnits
  paddings: ThemePaddings
}

/**
 * Inject the theme props into a styled component.
 */
export type ThemedProps<P> = ThemedStyledProps<P, Theme>

declare module 'styled-components/native' {
  export interface DefaultTheme extends Theme {}
}
