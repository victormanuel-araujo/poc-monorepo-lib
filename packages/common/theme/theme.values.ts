import { ThemeColorsGroup } from './theme.types'

/**
 * ** IMPORTANT **
 * Remember to always use the literal type in each value, this is because, during
 * the autocomplete, the majority of IDEs can show the value of de property on this ways.
 * If this is not defined, the IDE will show the type of property as string/number
 *
 * see an example:
 * - with literal type:
 *  https://i.imgur.com/n7piYD9.png
 *
 * - without literal type:
 * https://i.imgur.com/JZ9rcI8.png
 */

/**
 * Define the colors of theme
 * to add a new color in theme, remember to add the color name at ThemeColorsName
 * the shades are defined in ThemeColorsShade, remember: the base shade is the
 * only one needed
 */
export const colors: ThemeColorsGroup = {
  primary: {
    base: '#FF8C00' as '#FF8C00',
  },
  secondary: {
    base: '#7D38DB' as '#7D38DB',
  },
  black: {
    base: '#000000' as '#000000',
    light: '#202020' as '#202020',
    lighter: '#323232' as '#323232',
  },
  white: {
    base: '#FFFFFF' as '#FFFFFF',
  },
  gray: {
    base: '#9A9A9A' as '#9A9A9A',
    dark: '#888888' as '#888888',
    light: '#DADADA' as '#DADADA',
    lighter: '#E0E0E0' as '#E0E0E0',
  },
  red: {
    base: '#DB4446' as '#DB4446',
  },
  transparent: {
    base: '#00000000',
  },
  yellow: {
    base: '#FFBC11',
  },
}

/**
 * Define the name of the fonts families
 */
export const fonts = {
  /**
   * Used in common text elements
   */
  default: 'Raleway' as 'Raleway',
  /**
   * Used in Titles of sections, headers and etc.
   */
  title: 'Raleway' as 'Raleway',
}

/**
 * Define the fonts sizes
 * this values will be used as pixels (px)
 */
export const fontSizes = {
  xsm: 8 as 8,
  sm: 12 as 12,
  md: 16 as 16,
  lg: 24 as 24,
}

/**
 * Define the line heights for each font size.
 * IMPORTANT: these values will be used with the fontsSizes, so you must make sure
 * that: the values are sync as in the example
 *
 * in short, if an text element use the "md" of fontSizes, it must use the "md"
 * of lineHeights too
 *
 * @example
 * ```ts
 * styled.Text`
 *    font-size: ${fontSizes.md}
 *    line-height: ${fontSizes.md}
 * `
 * ```
 */
export const lineHeights = {
  xsm: 12 as 12,
  sm: 18 as 18,
  md: 24 as 24,
  lg: 36 as 36,
}

/**
 * Define the fonts weights
 */
export const fontWeights = {
  regular: 400 as 400,
  medium: 500 as 500,
  bold: 700 as 700,
  black: 900 as 900,
}

/**
 * Define the measure units of the theme
 * any values can be defined here, these values will be used as global within
 * the theme
 *
 * @example
 * ```ts
 * const units = {
 *   base: 8 (measure base)
 *   header: 60 (header height)
 *   navigation: 70 (navigation - tabstack - height)
 * }
 * ```
 */
export const units = {
  base: 8 as 8,
  header: 60 as 60,
  radius: 8 as 8,
}

/**
 * define the value of different sizes of padding
 * remember to always use the unit base defined before.
 */
export const paddings = {
  sm: units.base,
  md: (units.base * 2) as 16,
  lg: (units.base * 3) as 24,
  xl: (units.base * 4) as 32,
  xxl: (units.base * 5) as 40,
}
