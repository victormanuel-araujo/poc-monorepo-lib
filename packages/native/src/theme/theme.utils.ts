// utils functions to handle theme
import { get, defaultsDeep } from 'lodash'
import { Platform } from 'react-native'
import { css } from 'styled-components/native'

import { theme as _theme } from './theme'
import {
  Theme,
  ThemeColorsName,
  ThemeColorsShade,
  ThemeFontWeights,
  ThemeFontSizes,
  ThemePaddings,
  ThemeProps,
  ThemeFonts,
} from './theme.types'

/**
 * This function create an theme object.
 * Optionally, you can pass an custom theme object with custom values and it will
 * always be merged with the default theme (smash)
 *
 * @param theme - custom theme
 *
 * @example
 * ```ts
 * const theme_a = { a: 1, b: 2, c: 3 };
 * const theme_b = { a: 2 };
 *
 * const theme_c = createTheme(theme_a, theme_b);
 *
 * // the output (theme_c) is equal to:
 * // theme_c = { a: 2, b: 2, c: 3 };
 * ```
 */
export const createTheme = (theme?: DeepPartial<Theme>): Theme => {
  return defaultsDeep(theme, _theme)
}

/**
 * easy way to get an theme color with a shade and optionally an alpha channel
 *
 * @param colorName - the name of the color
 * @param colorShade - the shade of the color
 * @param alpha - an alpha value in hexadecimal pattern, that is: two character from '00' to 'FF'
 *
 * @example
 * ```ts
 * inside a styled-component
 * ${getThemeColor(...props)}
 *
 * outside a styled-component
 * getThemeColor(...props)({ theme })
 * ```
 */
export const getThemeColor =
  (colorName: ThemeColorsName, colorShade: keyof ThemeColorsShade = 'base', alpha?: string) =>
  ({ theme }): string => {
    const color = get(theme, ['colors', colorName], { base: '#000000' })
    const shade: string = get(color, [colorShade], color.base)

    return alpha && alpha.length === 2 ? `${shade}${alpha}` : shade
  }

/**
 * Get the style of a font weight.
 * Verify the platform where the component is used, and return te correct style
 * to apply the font weight
 *
 * @param weight - Font weight, ie light, regular, bold, etc.
 * @param family - Font family - only used on Android
 *
 * @example
 * ```ts
 * const Bold = styled.Text`
 *   ${getFontWeight('bold')}
 * `
 * ```
 */
export const getFontWeight =
  (weight: keyof ThemeFontWeights, family: keyof ThemeFonts = 'default') =>
  ({ theme }) => {
    /*
     * In android, is necessary to change the font family too, cause the
     * font-weight property not work correctly
     */
    if (Platform.OS === 'android') {
      /* get the font family based on "family" prop  */
      const fontFamily: string = get(theme, ['fonts', family])

      /* return the style, with font-family and font-weight defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
        font-weight: ${get(theme, ['fontWeights', weight])};
      `
    }

    /*
     * In another platforms, as IOS or Web, is not necessary to set a new font family
     * so,
     */
    return css`
      font-weight: ${theme.fontWeights[weight]};
    `
  }

/**
 * Get the correct style for a font size.
 * return a style, with the font size and the relative line-height.
 *
 * @param size - Font size name, ie sm, md, lg, etc.
 * @param rawValue - If it's true, returns the raw integer value
 *                   instead of a style.
 *
 * @example
 * ```ts
 * const Title = styled.Text`
 *   ${getFontSize('lg')};
 * `
 * ```
 *
 * @example - `theme` comes from styled-component's ThemeContext.
 * ```tsx
 *   return <Icon size={getFontSize('lg', true)({ theme })} />
 * ```
 */
export const getFontSize =
  (size: keyof ThemeFontSizes, rawValue = false) =>
  ({ theme }: ThemeProps<Theme>) => {
    if (rawValue) {
      return get(theme, ['fontSizes', size])
    }

    return css`
      font-size: ${get(theme, ['fontSizes', size])}px;
      line-height: ${get(theme, ['lineHeights', size])}px;
    `
  }

/**
 * Used to get a style of a font family and verify the platform where the
 * component is used, and return te correct style
 * to apply the font family
 *
 * @param family - the name of the family
 * @param weight - optionally an weight of the family. use it only in android
 *
 * *
 * @example
 * ```ts
 * const TitleFamily = styled.Text`
 *   ${getFontFamily('title')}
 * `
 * ```
 */
export const getFontFamily =
  (family: keyof ThemeFonts = 'default', weight: keyof ThemeFontWeights = 'medium') =>
  ({ theme }: ThemeProps<Theme>) => {
    /* get the font family based on "family" prop  */
    const fontFamily: string = get(theme, ['fonts', family])

    if (Platform.OS === 'android') {
      /* return the style, with font-family defined correctly */
      return css`
        font-family: ${fontFamily}-${weight};
      `
    }

    return css`
      font-family: ${fontFamily};
    `
  }

/**
 * Get a named padding from current theme.
 *
 * @param size - Padding size name, ie sm, md, lg, etc.
 * @param multiply - used if is necessary to multiply the padding value
 *
 * @example
 * ```ts
 * const Container = styled.View`
 *   padding: ${getPadding('md')} ${getPadding('lg')};
 * `
 * ```
 */
export const getPadding =
  (size: keyof ThemePaddings, multiply = 1) =>
  ({ theme }: ThemeProps<Theme>): string => {
    return theme.paddings[size] * multiply + 'px'
  }
