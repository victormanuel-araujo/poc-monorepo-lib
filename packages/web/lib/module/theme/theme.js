import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from "./theme.values";
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */

export const theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings
};
//# sourceMappingURL=theme.js.map