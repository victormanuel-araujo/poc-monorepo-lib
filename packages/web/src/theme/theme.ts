import { Theme } from './theme.types'
import { colors, fonts, fontSizes, fontWeights, lineHeights, paddings, units } from './theme.values'
/*
 * in the future, if exists a second theme, just have to create a second
 * composition as same is `theme`.
 */

/**
 * Default gama theme - SMASH
 */
export const theme: Theme = {
  colors,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  units,
  paddings,
}
