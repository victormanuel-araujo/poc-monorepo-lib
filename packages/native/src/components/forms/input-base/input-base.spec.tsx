import React from 'react'

import { Icon, InputBase } from '@components'
import { fireEvent } from '@testing-library/react-native'

import { renderWithTheme } from '../../../theme/theme.fixture'

describe('InputBase component', () => {
  it('should render correctly', () => {
    const component = renderWithTheme(<InputBase />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should exec the onChangeText function and pass the new value as param', () => {
    const func = jest.fn()
    const component = renderWithTheme(<InputBase onChangeText={func} />)
    const input = component.getByTestId('input-base')

    fireEvent.changeText(input, 'lorem ipsum')

    expect(func).toHaveBeenCalledWith('lorem ipsum')
  })

  it('should render the adornment correctly', () => {
    const component = renderWithTheme(<InputBase startAdornment={<Icon.Material icon="search" />} />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render the error type correctly', () => {
    const component = renderWithTheme(<InputBase error />)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should do not call the onChangeText function when input is disabled', () => {
    const func = jest.fn()
    const component = renderWithTheme(<InputBase onChangeText={func} disabled={true} />)
    const input = component.getByTestId('input-base')

    fireEvent.changeText(input, 'lorem ipsum')

    expect(func).not.toHaveBeenCalled()
  })

  it('should the type "password" have the "secureTextEntry" prop', () => {
    const component = renderWithTheme(<InputBase type="password" />)
    const input = component.getByTestId('input-base')

    expect(input.props.secureTextEntry).toBeTruthy()
  })

  it('should the "onFocus" function be called', () => {
    const func = jest.fn()
    const component = renderWithTheme(<InputBase onFocus={func} />)
    const input = component.getByTestId('input-base')

    fireEvent(input, 'focus')
    expect(func).toHaveBeenCalled()
  })

  it('should the "onBlur" function be called', () => {
    const func = jest.fn()
    const component = renderWithTheme(<InputBase onBlur={func} />)
    const input = component.getByTestId('input-base')

    fireEvent(input, 'blur')
    expect(func).toHaveBeenCalled()
  })

  it('should not duplicated accessibility label', () => {
    const component = renderWithTheme(<InputBase accessibilityLabel="Sample input component" />)

    expect(component.getAllByA11yLabel('Sample input component')).toHaveLength(1)
  })

  it('shoud work as a controlled input', () => {
    function Input() {
      const [text, setText] = React.useState('')

      // Remove numbers
      const handleChangeText = (newText: string) => setText(newText.replace(/\d/g, ''))

      return <InputBase value={text} onChangeText={handleChangeText} />
    }

    const component = renderWithTheme(<Input />)
    const internalTextInput = component.getByTestId('input-base')

    fireEvent.changeText(internalTextInput, 'abc123')

    expect(internalTextInput).toHaveProp('value', 'abc')
  })

  it('should avoid updates when value doesnt change on controlled components', () => {
    function Input() {
      const [text, setText] = React.useState('')

      // Accept only numbers
      const handleChangeText = (newText: string) => {
        if (/^\d+$/.test(newText)) {
          setText(newText)
        }
      }

      return <InputBase value={text} onChangeText={handleChangeText} />
    }

    const component = renderWithTheme(<Input />)
    const internalTextInput = component.getByTestId('input-base')

    fireEvent.changeText(internalTextInput, '123')
    fireEvent.changeText(internalTextInput, '123.')

    expect(internalTextInput).toHaveProp('value', '123')
  })

  it('should avoid updates when is not controlled', () => {
    const component = renderWithTheme(<InputBase />)
    const internalTextInput = component.getByTestId('input-base')

    fireEvent.changeText(internalTextInput, '123')
    fireEvent.changeText(internalTextInput, '123.')

    expect(internalTextInput).toHaveProp('value', '123.')
  })
})
