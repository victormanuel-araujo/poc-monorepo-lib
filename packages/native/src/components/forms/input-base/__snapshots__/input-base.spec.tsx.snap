// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase component should render correctly 1`] = `
<View
  accessible={true}
  fluid={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    onLayout={[Function]}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "minHeight": 54,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      backgroundShade="base"
      color="secondary"
      dir="column"
      display="flex"
      position="relative"
      style={
        Object {
          "backgroundColor": "#7D38DB",
          "borderRadius": 8,
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="input-base-shadow"
    />
    <View
      focused={false}
      spacing="md"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgba(218, 218, 218, 1)",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 2,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 2,
          "position": "relative",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        backgroundShade="base"
        dir="column"
        display="flex"
        position="relative"
        style={
          Object {
            "display": "flex",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          color="secondary"
          editable={true}
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
              Object {},
            ]
          }
          testID="input-base"
          type="text"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`InputBase component should render the adornment correctly 1`] = `
<View
  accessible={true}
  fluid={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    onLayout={[Function]}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "minHeight": 54,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      backgroundShade="base"
      color="secondary"
      dir="column"
      display="flex"
      position="relative"
      style={
        Object {
          "backgroundColor": "#7D38DB",
          "borderRadius": 8,
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="input-base-shadow"
    />
    <View
      focused={false}
      spacing="md"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgba(218, 218, 218, 1)",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 2,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 2,
          "position": "relative",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        alignment="center"
        backgroundShade="base"
        dir="column"
        display="flex"
        justifyContent="center"
        paddingBottom={0}
        paddingLeft={0}
        paddingTop={0}
        position="relative"
        spacing="sm"
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 8,
            "paddingTop": 0,
            "position": "relative",
            "width": "auto",
          }
        }
      >
        <View
          shape="round"
          size={24}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#000000",
                "borderRadius": 32,
                "display": "flex",
                "height": 32,
                "justifyContent": "center",
                "width": 32,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¢¶
          </Text>
        </View>
      </View>
      <View
        backgroundShade="base"
        dir="column"
        display="flex"
        position="relative"
        style={
          Object {
            "display": "flex",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          color="secondary"
          editable={true}
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
              Object {},
            ]
          }
          testID="input-base"
          type="text"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`InputBase component should render the error type correctly 1`] = `
<View
  accessible={true}
  fluid={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      },
      false,
      undefined,
    ]
  }
>
  <View
    backgroundShade="base"
    dir="column"
    display="flex"
    onLayout={[Function]}
    position="relative"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "minHeight": 54,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      backgroundShade="base"
      color="secondary"
      dir="column"
      display="flex"
      error={true}
      position="relative"
      style={
        Object {
          "backgroundColor": "#DB4446",
          "borderRadius": 8,
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "left": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="input-base-shadow"
    />
    <View
      focused={false}
      spacing="md"
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DB4446",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 2,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 2,
          "position": "relative",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        backgroundShade="base"
        dir="column"
        display="flex"
        position="relative"
        style={
          Object {
            "display": "flex",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          color="secondary"
          editable={true}
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
              Object {},
            ]
          }
          testID="input-base"
          type="text"
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
</View>
`;
